### PR TITLE
Feature reset

### PR DESCRIPTION
**Pull request by Anas12091101**

### CODE REVIEW

File: ScratchDjango/User/views.py [hunk 1]
```diff
@@ -39,7 +39,9 @@ def register_user(request):
         )
 
         message = f"Hi {user.email}, Welcome to DjangoFromScratch. We hope you enjoy our product and have a good time here."
+        print(data["otp_enabled"])
```
Comment (zamanafzal) [inline ScratchDjango/User/views.py:42]: Please remove all the print statements

File: ScratchDjango/User/views.py [hunk 2]
```diff
@@ -53,7 +55,7 @@ def register_user(request):
 def login_user(request):
     user = authenticate(email=request.data["email"], password=request.data["password"])
     if user:
-        if not user.otp_enabled:
+        if user.otp_enabled=="false":
```
Comment (zamanafzal) [inline ScratchDjango/User/views.py:58]: I don't think this will return the right result.

File: ScratchDjango/User/forms.py [hunk 3]
```diff
@@ -14,6 +15,22 @@ class Meta:
...
+
+    class Meta:
+        model = User
+        fields = ("password", "confirm_password")
+
+
+class ResetEmailForm(forms.ModelForm):
+    class Meta:
+        model = User
+        fields = ["email"]
```
Comment (zamanafzal) [inline ScratchDjango/User/forms.py:36]: I think it would be better to add clean_field method here.

File: ScratchDjango/User/views.py (hunk 2 above)
Comment (Anas12091101) [reply ScratchDjango/User/views.py:58]: Oh yes, I was testing it with the django templates login page and it was returning false for no value selected. Forgot to change it for react. Will change it

File: ScratchDjango/User/forms.py (hunk 3 above)
Comment (Anas12091101) [reply ScratchDjango/User/forms.py:36]: Ok

File: ScratchDjango/User/views.py (hunk 1 above)
Comment (Anas12091101) [reply ScratchDjango/User/views.py:42]: Ok

File: ScratchDjango/User/views.py [hunk 4]
```diff
@@ -180,3 +181,44 @@ def otp_template(request, email):
...
+
+def confirm_reset_template(request, token):
+    if request.method == "POST":
+        password = request.POST["password"]
+        confirm = request.POST["confirm_password"]
+        try:
+            if password != confirm:
+                raise ValidationError("Password donot match")
+            url = f"{HOST}/user/api/password_reset/confirm/"
+            payload = json.dumps({"token": token, "password": password})
```
[the remark this answers is not in the document]
Comment (Anas12091101) [reply ScratchDjango/User/views.py:212]: This is the template view used before we created the react application. I will remove all of them in the next commit.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+```bash
+  cd ScratchDjango
+```
+
+Add .env file
+
+```bash
+  touch .env
+  // Following variables should be set in environment
```
Comment (zamanafzal) [inline README.md:30]: Can we have dummy values here.

**Review by zamanafzal — APPROVED** [review]
Overall looks good. just add sample values in readme file

File: README.md (hunk 5 above)
Comment (Anas12091101) [reply README.md:30]: yes, updating..

**Pull request merged**